app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt

```diff
@@ -90,10 +90,14 @@ class ChatSettingsFragment : PreferenceFragmentCompat(), ISettingsView {
 
         // Display login email
         val utilModel = UtilModel(activity as SkillsActivity)
-        if (!utilModel.isLoggedIn())
+        if (!utilModel.isLoggedIn()) {
```

Comment:
Shouldn't the Presenter be interacting with the model and not the view? @batbrain7

Reply:
this code has been touched for the first time since last year @chiragw15 , much refactoring is needed, so this will be done in consequent issues or if it needs to be done now I'll refactor this Fragment and then update the PR ?

Reply:
@batbrain7 Should it check for utilModel.isLoggedIn() or should it check for null acess token i.e. ```if(PrefManager.getToken() != null)``` for checking if the user is logged in because that automatically takes care of token expiry and also considers the case when user skips login?

Reply:
thanks @arundhati24 I will use this instead